### ISSUE TITLE
pyside2 as qt5 backend

### ISSUE THREAD

**Issue by neok-m4700** [matplotlib/matplotlib#8618]
### Bug report

**Bug summary**
using Pyside2 as a qt5 backend is failing

**Code for reproduction**
```python
import matplotlib; matplotlib.use('Qt5Agg'); matplotlib.rcParams['backend.qt5']='PySide2'
import matplotlib.pyplot as plt
fig, ax = plt.subplots()
```

**Actual outcome**
```python
    fig, ax = plt.subplots()
  File "<...>/lib/python3.6/site-packages/matplotlib/pyplot.py", line 1189, in subplots
    fig = figure(**fig_kw)
  File "<...>/lib/python3.6/site-packages/matplotlib/pyplot.py", line 544, in figure
    **kwargs)
  File "<...>/lib/python3.6/site-packages/matplotlib/backends/backend_qt5agg.py", line 44, in new_figure_manager
    return new_figure_manager_given_figure(num, thisFig)
  File "<...>/lib/python3.6/site-packages/matplotlib/backends/backend_qt5agg.py", line 51, in new_figure_manager_given_figure
    canvas = FigureCanvasQTAgg(figure)
  File "<...>/lib/python3.6/site-packages/matplotlib/backends/backend_qt5agg.py", line 246, in __init__
    super(FigureCanvasQTAgg, self).__init__(figure=figure)
  File "<...>/lib/python3.6/site-packages/matplotlib/backends/backend_qt5agg.py", line 68, in __init__
    super(FigureCanvasQTAggBase, self).__init__(figure=figure)
  File "<...>/lib/python3.6/site-packages/matplotlib/backends/backend_qt5.py", line 246, in __init__
    super(FigureCanvasQT, self).__init__(figure=figure)
AttributeError: 'figure()' is not a Qt property or a signal
```

**Matplotlib version**

  * Operating System: Ubuntu 17.04
  * Matplotlib Version: f1072075db6615954cac2ff0fa44c4d6c7764a23 (latest commit on master)
  * Python Version: 3.6.1
  * Other Libraries: Qt 5.8.0 + Pyside2

what can I do ?

**Comment by tacaswell** [comment]
Is pyside2 not cooperative?  If not we need to run the it through the qt4Agg code path instead of directly through the qt5 code path.

**Comment by neok-m4700** [comment]
Hi, thanks for the clues, I'll dig into the pyside2 API.

**Comment by neok-m4700** [comment]
I have made progress and was able to build matplotlib with qt4/pyside and qt5/pyside2

Here is the patch: https://github.com/neok-m4700/matplotlib/commit/13a5d5ba90ea3e4408a1c72d934dd8d15973f2da

I used qtpy from spyder as proposed in https://github.com/matplotlib/matplotlib/issues/6118

I ran the test suite python -c 'import matplotlib; matplotlib.test()', and it passed without issues on this part.

Should I make a pull request ?

**Comment by tacaswell** [comment]
Yes please, but it will need a few more modifications.

 - if we are going to us qtpy then we should delete our entire compatibility shim and add it as a hard dependency 
 - I am not a fan of the `if` statement in the init, either we should fully merge the qt4/qt5 code bases or run the qt5 + pyside2 through the qt4 code paths (which already account for the non-cooperative super classes

This probably crosses paths with https://github.com/matplotlib/matplotlib/issues/8613

Is there a reasonable path where we have a shim backend called 'qtagg' which imports from `qt_compat` and then decides if it needs to use the cooperative or non-cooperative versions of the classes?

It is really disappointing that pyside2 is not cooperative :disappointed: .

**Comment by anntzer** [comment]
Closed by #9052.
